api/rql/internal/errz/matchError.go

```diff
@@ -0,0 +1,38 @@
+package errz
+
+import "fmt"
+
+// MatchError represents the case when the input tokens did not
+// match a given node
+type MatchError struct {
+	reason string
+}
+
+func (m MatchError) Error() string {
+	return m.reason
+}
+
+// MatchErrorf creates a new MatchError object
+func MatchErrorf(format string, a ...interface{}) MatchError {
+	return MatchError{fmt.Sprintf(format, a...)}
+}
+
+// IsMatchError returns true if err is a MatchError,
+// false otherwise.
+func IsMatchError(err error) bool {
```

Comment:
You can just use `errors.As(err, MatchError{})` in Go 1.13.

Reply:
Hmm, that doesn't quite work because it returns true if `err`'s chain contains first instance of MatchError. Some parts of the RQL code wrap match errors, see e.g. https://github.com/ekinanp/wash/blob/rql/api/rql/internal/predicate/numeric.go#L121, so that they could be returned as syntax errors. There, `errors.As` will return true.

I think `errors.As`/`errors.Is` should have been named to something like `errors.Contains`. Since `Contains` is different from `IsMatchError`, I'm opting to keep the helper.

I did find that `IsSyntaxError` is not being used anywhere, so I'll go ahead and remove that.

Reply:
Updated.